docs/admin/user-management.rst

```diff
@@ -1,34 +1,165 @@
 .. _administration_user_management:
 
-===============
-User management
-===============
+==========================
+Users and roles management
+==========================
 
-User account information is stored in the cluster metadata of CrateDB and
-supports the following statements to create, alter and drop users:
+Users and roles account information is stored in the cluster metadata of CrateDB
+and supports the following statements to create, alter and drop users and roles:
 
     * `CREATE USER`_
-    * `ALTER USER`_
-    * `DROP USER`_
+    * `CREATE ROLE`_
+    * `ALTER USER`_ or `ALTER ROLE`_
+    * `DROP USER`_ or `DROP ROLE`_
 
 These statements are database management statements that can be invoked by
-superusers that already exist in the CrateDB cluster. The `CREATE USER`_ and
-`DROP USER`_ statements can also be invoked by users with the ``AL`` privilege.
+superusers that already exist in the CrateDB cluster. The `CREATE USER`_,
+`CREATE ROLE`_, `DROP USER`_ and `DROP ROLE`_ statements can also be invoked by
+users with the ``AL`` privilege. `ALTER USER`_ or `ALTER ROLE`_ can be invoked
+by users to change their own password, without requiring any privilege.
 
 When CrateDB is started, the cluster contains one predefined superuser. This
-user is called ``crate``. It is not possible to create any other superusers
+user is called ``crate``. It is not possible to create any other superusers.
 
 The definition of all users and roles, including hashes of their passwords,
-together with their privileges is backed up together with the cluster's metadata
-when a snapshot is created, and it is restored when using the ``ALL``,
-``METADATA``, or ``USERMANAGEMENT`` keywords with the
-:ref:`sql-restore-snapshot` command.
+together with their :ref:`privileges <administration-privileges>` is backed up
+together with the cluster's metadata when a snapshot is created, and it is
+restored when using the ``ALL``, ``METADATA``, or ``USERMANAGEMENT`` keywords
+with the:ref:`sql-restore-snapshot` command.
 
 .. rubric:: Table of contents
 
 .. contents::
    :local:
 
+``ROLES``
+---------
+
+Roles are entities that are **not** allowed to login, but can be assigned
+privileges and they can be granted to other roles, thus creating a role
+hierarchy, or directly to users. For example, a role ``myschema_dql_role`` can
+be granted with ``DQL`` privileges on schema ``myschema`` and afterwards the
+role can be :ref:granted`<granting_roles>` to a user, which will automatically
+ref:inherit`<roles_inheritance>` those privileges from the
+``myschema_dql_role``. A role ``myschema_dml_role`` can be granted with ``DML``
+privileges on schema ``myschema`` and can also be granted the role
+``myschema_dql_role``, thus gaining also ``DQL`` privileges. When
+``myschema_dml_role`` is granted to a user, this user will automatically have
+both ``DQL`` and ``DML`` privileges on ``myschema``.
```

Comment:
There are a few references that don't render for me:

<img width="857" alt="Screenshot 2023-12-29 at 15 08 02" src="https://github.com/crate/crate/assets/218003/eaf918b8-ad59-47ef-8217-e0a70dfcc658">